sfall/AnimationsAtOnceLimit.cpp

```diff
@@ -143,160 +143,181 @@ static const DWORD sad_28[] = {
-
-	//struct array 1///////////////////
-
-	//old addr 0x54C1B4
-	SafeWrite32(0x413A9E, (DWORD)anim_set);
-
-	//old addr 0x54C1C0
-	for (i = 0; i < sizeof(fake_anim_set_C)/4; i++) {
-		SafeWrite32(fake_anim_set_C[i], 12+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC14
-	for (i = 0; i < sizeof(anim_set_0)/4; i++) {
-		SafeWrite32(anim_set_0[i], 2656+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC18
-	for (i = 0; i < sizeof(anim_set_4)/4; i++) {
-		SafeWrite32(anim_set_4[i], 2656+4+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC1C
-	for (i = 0; i < sizeof(anim_set_8)/4; i++) {
-		SafeWrite32(anim_set_8[i], 2656+8+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC20
-	for (i = 0; i < sizeof(anim_set_C)/4; i++) {
-		SafeWrite32(anim_set_C[i], 2656+12+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC24
-	for (i = 0; i < sizeof(anim_set_10)/4; i++) {
-		SafeWrite32(anim_set_10[i], 2656+16+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC28
-	for (i = 0; i < sizeof(anim_set_14)/4; i++) {
-		SafeWrite32(anim_set_14[i], 2656+20+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC38
-	SafeWrite32(0x413F29, 2656+36+(DWORD)anim_set);
-
-	//old addr 0x54CC3C
-	for (i = 0; i < sizeof(anim_set_28)/4; i++) {
-		SafeWrite32(anim_set_28[i], 2656+40+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC48
-	SafeWrite32(0x415C35, 2656+52+(DWORD)anim_set);
-
-
-	//struct array 2///////////////////
-
-	//old addr 0x530014
-	for (i = 0; i < sizeof(sad_0)/4; i++) {
-		SafeWrite32(sad_0[i], (DWORD)sad);
-	}
-
-	//old addr 0x530018
-	for (i = 0; i < sizeof(sad_4)/4; i++) {
-		SafeWrite32(sad_4[i], 4+(DWORD)sad);
-	}
-
-	//old addr 0x53001C
-	for (i = 0; i < sizeof(sad_8)/4; i++) {
-		SafeWrite32(sad_8[i], 8+(DWORD)sad);
+static void __declspec(naked) anim_set_end_hook() {
+	__asm {
+		mov  edi, _anim_set
+		cmp  dword ptr AnimationsLimit, 32
+		jbe  skip
+		mov  edi, anim_set
+		add  edi, 2656                            // Adding a pacifier
+skip:
+		test dl, 0x2                              // Flag of combat mode?
+		jz   end                                  // No
+		call combat_anim_finished_
+end:
+		mov  [edi][esi], ebx
+		push 0x415DF2
+		retn
 	}
+};
 
-	//old addr 0x530020
-	for (i = 0; i < sizeof(sad_C)/4; i++) {
-		SafeWrite32(sad_C[i], 12+(DWORD)sad);
-	}
+void AnimationsAtOnceInit() {
 
-	//old addr 0x530024
-	for (i = 0; i < sizeof(sad_10)/4; i++) {
-		SafeWrite32(sad_10[i], 16+(DWORD)sad);
-	}
+	AnimationsLimit = GetPrivateProfileIntA("Misc", "AnimationsAtOnceLimit", 32, ini);
+	if (AnimationsLimit > 32) {
+		if (AnimationsLimit > 127) AnimationsLimit = 127;
+		dlog("Applying AnimationsAtOnceLimit patch.", DL_INIT);
+		int aniMax = AnimationsLimit; 	
+		int i;
 
-	//old addr 0x530028
-	for (i = 0; i < sizeof(sad_14)/4; i++) {
-		SafeWrite32(sad_14[i], 20+(DWORD)sad);
-	}
+		//allocate memory to store larger animation struct arrays
+		anim_set = new BYTE[2656*(aniMax+1)];
+		sad = new BYTE[3240*(aniMax+1)]; // here (aniMax-8) from @Crafty. Bug? 
```

Comment:
No, the (aniMax-8) has nothing to do with file size optimization.  Guess you didn't try to compile with aniMax-8 and aniMax+1 and check the file size.

Before the whole AnimationsAtOnceLimit.cpp contains lots of individual SafeWriteX(), and Crafty replaced them with grouped arrays and for loops in current form at Feb 9 2016 to reduce the file size. We've already implemented the change 1 year ago (commit 56009e1634c67c12c7a7a8bac1e43a3726e68597)